### PR TITLE
fix!: update to the new runner registration system

### LINKED ISSUE #28: Implement alternative to resolve upstream flavor package's dependency on the registry1 gitlab-runner-helper image.
### Describe what should be investigated or refactored
The upstream helper image currently does not work correctly in our locked down deployment configuration. This means that we have had to use the registry1 helper in the upstream deployment which limits it from truly being multi-arch. 

### Links to any relevant code
https://github.com/defenseunicorns/uds-package-gitlab-runner/blob/9025c6bf550361ec7c93b1ec623d8c14a74e2c5b/zarf.yaml#L73

### Additional context
Add any other context or screenshots about the technical debt here.

### LINKED ISSUE #74: Update GitLab runner to use new registration token system
### Describe what should be investigated or refactored
The current way that we register runners is deprecated and will be removed in a future release of GitLab - we should update this to the new model and consider ways that we can improve the registration experience per https://github.com/defenseunicorns/uds-package-gitlab-runner/issues/66 (if possible)

### Links to any relevant code
https://github.com/defenseunicorns/uds-package-gitlab-runner/issues/66

### Additional context
This registration method should be available until v18 under feature flags but we should update this to avoid potential issues in the future.

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This updates the registration system to the new runner authentication token (counting it as a bug fix as users have observed manual steps being needed in the current workflow).

## Related Issue

Fixes #74
Fixes #28 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-package-gitlab-runner/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: zarf.yaml [hunk 1]
```diff
@@ -37,5 +37,5 @@ components:
           - values/upstream-values.yaml
     images:
       - "registry.gitlab.com/gitlab-org/gitlab-runner:alpine-v17.0.0"
-      - "registry1.dso.mil/ironbank/gitlab/gitlab-runner/gitlab-runner-helper:v17.0.0"
+      - "registry.gitlab.com/gitlab-org/ci-cd/gitlab-runner-ubi-images/gitlab-runner-helper-ocp:v17.0.0"
```
Comment (zachariahmiller) [inline zarf.yaml:40]: ❤️

**Comment by Racer159** [comment]
New test regimen adds ~1 min 
image

File: tasks/test.yaml [hunk 2]
```diff
@@ -1,4 +1,30 @@
...
+      - task: glr-backup-registration-secret
+      # Remove the GLR package and redeploy with the manual token
+      - task: glr-remove
+      # TODO: (@WSTARR) Maru will complain about "cyclical" task imports if this imports the deploy task from uds-common.  This is a bug: https://github.com/defenseunicorns/maru-runner/issues/122
+      - description: Get the current UDS Bundle name
+        cmd: cat bundle/uds-bundle.yaml | yq .metadata.name
+        setVariables:
+          - name: BUNDLE_NAME
+      - description: Get the current UDS Bundle version
+        cmd: cat bundle/uds-bundle.yaml | yq .metadata.version
```
Comment (zachariahmiller) [inline tasks/test.yaml:19]:
```suggestion
        cmd: cat bundle/uds-bundle.yaml | ./uds zarf tools yq .metadata.version
```

File: tasks/test.yaml [hunk 3]
```diff
@@ -1,4 +1,30 @@
...
+      - task: glr-health-check
+      - task: glr-run-check
+      # Create a runner token and hide the secret from the GLR package
+      - task: glr-create-runner-token
+      - task: glr-backup-registration-secret
+      # Remove the GLR package and redeploy with the manual token
+      - task: glr-remove
+      # TODO: (@WSTARR) Maru will complain about "cyclical" task imports if this imports the deploy task from uds-common.  This is a bug: https://github.com/defenseunicorns/maru-runner/issues/122
+      - description: Get the current UDS Bundle name
+        cmd: cat bundle/uds-bundle.yaml | yq .metadata.name
```
Comment (zachariahmiller) [inline tasks/test.yaml:15]:
```suggestion
        cmd: cat bundle/uds-bundle.yaml | ./uds zarf tools yq .metadata.name
```

File: tasks/test.yaml [hunk 4]
```diff
@@ -25,3 +51,44 @@ tasks:
...
+            --form "token=${RUNNER_REGISTRATION_TOKEN}" --form "tag_list=uds" \
+            | ./uds zarf tools yq .token
+        mute: true
+        setVariables:
+          - name: RUNNER_AUTH_TOKEN
+
+  - name: glr-backup-registration-secret
+    description: Backup the registration secret and remove the original (to ensure it cannot be used by the package)
+    actions:
+      - cmd: ./uds zarf tools kubectl get secret -n gitlab gitlab-gitlab-runner-secret -o yaml | ./uds zarf tools yq 'del(.metadata.namespace,.metadata.resourceVersion,.metadata.uid) | .metadata.creationTimestamp=null' | kubectl apply --namespace=default -f -
```
Comment (zachariahmiller) [inline tasks/test.yaml:87]: raw kubectl in here and others

**Review by zachariahmiller — CHANGES_REQUESTED** [review]

**Comment by Racer159** [comment]
Added breaking change since this now requires UDS Core [v0.22.0](https://github.com/defenseunicorns/uds-core/releases/tag/v0.22.0)

**Review by ericwyles — APPROVED** [review]

**Review by zachariahmiller — APPROVED** [review]

**Pull request merged**